vignettes/Vaccines.Rmd

```diff
@@ -32,161 +115,279 @@ starting_EIR <- 20
+output_control <- run_simulation(timesteps = sim_length * 2, parameters = simparams)
 ```
 
-Then we can run the simulation for a variety of Vaccination strategies:
+Then we can run the simulation for a variety of vaccination strategies.
 
 ## Mass RTS,S
 
-This is a round of RTS,S vaccine for individuals between 5 months and 10 years followed by a booster after 18 months:
+First, we will model a single round of RTS,S vaccine given in a mass vaccination strategy where those within a given age range will be vaccinated to a specified coverage level and during a short window of time. In this example, individuals aged between 5 and 17 months are vaccinated with a primary series (3 doses) followed by a single booster 12 months after the initial vaccination. The age groups to be vaccinated as well as timing of all vaccinations can be modified within the `set_mass_rtss()` function.
 
-```{r}
-rtssparams <- simparams
+We specify that the intervention should use the RTS,S vaccine through the `profile` and `booster_profile` inputs.
 
+```{r}
 rtssparams <- set_mass_rtss(
-  rtssparams,
-  timesteps = 1 * year,
-  coverages = 1,
-  min_wait = 0,
-  min_ages = 5 * month,
-  max_ages = 10 * year,
-  boosters = 18 * month,
-  booster_coverage = 0.95
+  simparams,
+  timesteps = 1 * year, # The single round of vaccination is at 1 year into the simulation.
+  coverages = 1, # The vaccine is given to 100% of the population between the specified ages.
+  min_wait = 0, # The minimum acceptable time since the last vaccination is 0 because in our case we are only implementing one round of vaccination.
+  min_ages = 5 * month, # The minimum age for the target population to be vaccinated. 
+  max_ages = 5 * year, # The maximum age for the target population to be vaccinated.
+  boosters = 12 * month, # The booster is given at 12 months after the primary series. 
+  booster_coverage = 0.95 # Coverage of the booster dose is 95%.
 )
 
-output <- run_simulation(sim_length, rtssparams)
-output$clinical_incidence <- 1000 * output$n_inc_0_1825 / output$n_0_1825
+output <- run_simulation(timesteps = sim_length, parameters = rtssparams)
+```
+
+#### Plot clinical incidence
+
+```{r, fig.width = 7, fig.height = 5, fig.align = "center"}
+plot_inci()
+```
+
+#### Plot prevalence
+```{r, fig.width = 7, fig.height = 5, fig.align = "center"}
+plot_prev()
+```
+
+#### Plot doses
 
-ggplot(data = output, aes(x = timestep / 365, y = clinical_incidence)) + 
-  geom_line(col = "grey80") +
-  stat_smooth(col = "darkblue", se = FALSE) +
-  geom_vline(xintercept = 1, col = "red") +
-  xlab("year") +
-  ylab("Clinical incidence \n (per 1000 children aged 0-5)") +
-  theme_bw()
+You can look at the distribution of doses using the `n_rtss_mass_dose_*` or `n_rtss_epi_dose_*` outputs. It is always a good idea to check the timing of vaccine doses to ensure that it is specified as intended.
 
+```{r, fig.width = 7, fig.height = 5, fig.align = "center", message=FALSE}
+plot_doses()
```

Comment:
This errors for me:
```
Error in `[.data.frame`(output, , c(2:5, 38)) :
  undefined columns selected
```

Reply:
Sorry about that - I changed it so now it is selecting columns by their names, maybe that will help? It wasn't erroring for me, so it's a bit hard to check